src/components/ReplaceServerDetails.js

```diff
@@ -0,0 +1,321 @@
+    //if it is an existing server discovered use that uid
+    this.data['uid'] = this.state.replaceData.get('uid');
+    let theProps = {
+      wipeDisk : this.state.isWipeDiskChecked,
+      installOS : this.state.isInstallOsChecked
+    };
+    if(this.state.isInstallOsChecked) {
+      theProps.osInstallUsername = this.state.osInstallUsername;
+      theProps.osInstallPassword = this.state.osInstallPassword; //for now TODO
+    }
+    //picked from available server
+    if(this.state.selectedServerId) {
+      theProps.selectedServerId = this.state.selectedServerId;
+    }
+
+    this.props.doneAction(this.data, theProps);
+  }
+
+  handleCancel = () => {
+    this.props.cancelAction();
+  }
+
+  handleInputChange = (e, isValid, props) => {
+    //deselect the selected available server when
+    //user start entering values by himself
+    this.state.selectedServerId = '';
+    this.updateFormValidity(props, isValid);
+    let value = e.target.value;
+    if (isValid) {
+      if(props.inputName !== 'osInstallPassword') {
+        let newData = this.state.replaceData.set(props.inputName, value);
+        this.setState({replaceData: newData});
+      }
+      else {
+        this.setState({osInstallPassword: value}, () => {
+          this.setState({isFormValid: this.isFormTextInputValid()});
+        });
+      }
+    }
+  }
+
+  updateAutoFillFormValidity = (newJSData) => {
+    for (let key in newJSData) {
+      if(newJSData[key]) {
+        this.allInputsStatus[key] = INPUT_STATUS.VALID;
+      }
+      else {
+        this.allInputsStatus[key] = INPUT_STATUS.INVALID;
+      }
+    }
+    this.setState({isFormValid: this.isFormTextInputValid()});
+  }
+
+  handleOsPasswordChange = (e) => {
+    const password = e.target.value;
+    this.setState({osInstallPassword: password}, () => {
+      this.setState({isFormValid: this.isFormTextInputValid()});
+    });
+  }
+
+  handleInstallOsCheck = () => {
+    this.setState({
+      isInstallOsChecked: !this.state.isInstallOsChecked }, () => {
+      this.setState({isFormValid: this.isFormTextInputValid()});
+    });
+  }
+
+  handleWipeDiskCheck = () => {
+    this.setState({isWipeDiskChecked: !this.state.isWipeDiskChecked});
+  }
+
+  handleSelectAvailableServer = (serverId) => {
+    //TODO need to find a way to show the details of the available server
+    //selected
+    this.setState({selectedServerId: serverId});
+    let allAvailableServers = [];
+    if(this.props.autoServers && this.props.autoServers.length > 0) {
+      allAvailableServers = this.props.autoServers;
+    }
+    if(this.props.manualServers && this.props.manualServers.length > 0) {
+      allAvailableServers = allAvailableServers.concat(this.props.manualServers);
+    }
+
+    let theServer = allAvailableServers.find(server => server.id === serverId);
+    let newJSData = {};
+    if(theServer) {
+      newJSData['ilo-ip'] = theServer['ilo-ip'] ? theServer['ilo-ip'] : '';
```

Comment:
While this works, it is more straightforward to use:
```
newJSData['ilo-ip'] = theServer['ilo-ip'] || '';
```
on this line and the ones that follow

Reply:
will update